src/account.c

```diff
@@ -1720,83 +1720,27 @@ const char *account_extra(const struct account *acc)
 int account_uri_complete(const struct account *acc, struct mbuf *buf,
 			 const char *uri)
 {
-	struct sa sa_addr;
-	size_t len;
-	bool uri_is_ip;
-	char *uridup;
-	char *host;
-	char *c;
-	int err = 0;
+	char hostport[128];
+	char port[8];
+	bool print_port;
+	struct uri *luri;
 
 	if (!buf || !uri)
 		return EINVAL;
 
-	/* Skip initial whitespace */
-	while (isspace(*uri))
-		++uri;
-
-	len = str_len(uri);
-
-	/* Append sip: scheme if missing */
-	if (0 != re_regex(uri, len, "sip:"))
-		err |= mbuf_printf(buf, "sip:");
-
-	err |= mbuf_write_str(buf, uri);
-
-	if (!acc || err)
-		return err;
-
-	/* Append domain if missing and uri is not IP address */
-
-	/* check if uri is valid IP address */
-	err = str_dup(&uridup, uri);
-	if (err)
-		return err;
-
-	if (!strncmp(uridup, "sip:", 4))
-		host = uridup + 4;
-	else
-		host = uridup;
-
-	c = strchr(host, ';');
-	if (c)
-		*c = 0;
-
-	uri_is_ip =
-		!sa_decode(&sa_addr, host, strlen(host)) ||
-		!sa_set_str(&sa_addr, host, 0);
-	if (!uri_is_ip && host[0] == '[' && (c = strchr(host, ']'))) {
-		*c = 0;
-		uri_is_ip = !sa_set_str(&sa_addr, host+1, 0);
-	}
-
-	mem_deref(uridup);
-
-	if (0 != re_regex(uri, len, "[^@]+@[^]+", NULL, NULL) &&
-		1 != uri_is_ip) {
-#if HAVE_INET6
-		if (AF_INET6 == acc->luri.af)
-			err |= mbuf_printf(buf, "@[%r]",
-					   &acc->luri.host);
-		else
-#endif
-			err |= mbuf_printf(buf, "@%r",
-					   &acc->luri.host);
+	if (!acc || acc->regint == 0)
+		return uri_complete(uri, NULL, buf);
 
-		/* Also append port if specified and not 5060 */
-		switch (acc->luri.port) {
+	luri = account_luri(acc);
+	print_port = luri->port != SIP_PORT && luri->port != 0;
+	re_snprintf(port, sizeof(port), "%d", luri->port);
 
-		case 0:
-		case SIP_PORT:
-			break;
+	re_snprintf(hostport, sizeof(hostport), "%r%s%s",
+		    &luri->host,
+		    print_port ? ":" : "",
+		    print_port ? port : "");
 
-		default:
-			err |= mbuf_printf(buf, ":%u", acc->luri.port);
-			break;
-		}
-	}
-
-	return err;
+	return uri_complete(uri, hostport, buf);
```

Comment:
I dont think it matters where the function `uri_complete` is located.

The logical behaviour would be the same, even if located in A or B or C.